demo/dismissable/dismissable-item.js

```diff
@@ -0,0 +1,216 @@
+        break;
+      case 'touchend':
+        this._onPointerUp(event, event.changedTouches[0]);
+        break;
+    }
+  }
+
+  setPosition(position) {
+    this.position = position;
+    this.width = this.offsetWidth;
+    this.style.opacity = (this.width - Math.abs(position)) / this.width;
+    const currentY = this.style.transform.split(',')[1];
+    this.style.transform = `translate(${position}px, ${currentY}`;
+  }
+
+  _dismiss() {
+    this.style.opacity = 0;
+
+    const collapseAnim = this.animate({
+      height: [ getComputedStyle(this).height, '0px']
+    }, {
+      duration: 100,
+      iterations: 1
+    });
+
+    collapseAnim.onfinish = this._fireRemove.bind(this);
+  }
+
+  _fireRemove() {
+    this.setPosition(0);
+
+    const event = new CustomEvent('remove', {
+      detail: { itemIndex: this.itemIndex },
+      bubbles: true
+    });
+    this.dispatchEvent(event);
+  }
+
+  settle(targetPosition) {
+    this.state = 'initial';
+    if (targetPosition === this.position) {
+      return;
+    }
+
+    const currentY = this.style.transform.split(',')[1];
+    const isDismiss = targetPosition !== 0;
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, isDismiss ? 0 : 1 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) * 0.5,
+      iterations: 1
+    });
+
+    this.position = targetPosition;
+    animation.onfinish = () => isDismiss ? this._dismiss() : this.setPosition(0);
+  }
+
+  fling(velocityX) {
+    this.state = 'initial';
+    const targetPosition = velocityX < 0 ? -this.width : this.width;
+
+    const currentY = this.style.transform.split(',')[1];
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, 0 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) / Math.abs(velocityX),
+      iterations: 1
+    });
+
+    animation.onfinish = this._dismiss.bind(this);
+  }
+
+  _settleToClosestPosition() {
+    const fraction = this.position / this.width;
+    if (fraction > 0.5) {
+      this.settle(this.width);
+    } else if (fraction < -0.5) {
+      this.settle(-this.width);
+    } else {
+      this.settle(0);
+    }
+  }
+
+  _onPointerDown(e, change) {
+    this.state = 'initial';
+    this.startX = change.clientX;
+    this.startY = change.clientY;
+    this.startPosition = 0;
+  }
+
+  _onPointerPan(e, change) {
+    if (this.state == 'initial') {
+      const deltaX = change.clientX - this.startX;
+      const deltaY = change.clientY - this.startY;
+
+      // Shaky or sloppy fingers are common using scroll, so
+      // we ignore mini pans of 5 dips like Android.
+      if (deltaX ** 2 + deltaY ** 2 < kTouchSlopValue ** 2) {
+        e.preventDefault();
```

Comment:
not clear why we preventDefault here 🤔 
This one or the one below seem to cause browser intervention when scrolling vertically more than horizontally, e.g. 
![scroll-error](https://user-images.githubusercontent.com/6173664/41375256-59043136-6f0a-11e8-83b3-9fca41c62acd.gif)

This prompted the earlier suggestion to check if we're scrolling horizontally or not, and avoid updating position/opacity if the user is scrolling vertically.

Reply:
The second preventDefault avoid the following:

https://www.youtube.com/watch?v=f5KsaMW4RW4

Maybe we can fix that a better way

Reply:
oh interesting...I think that _should_ be fixable by setting
```css
virtual-scroller {
  overflow-x: hidden;
}

Reply:
The first perventDefaults can be removed.

I did run into this other bug though, which I have to find out why happens:

https://drive.google.com/file/d/1irkYq-XyI7C-HLedtXlms12D-cJfo1AX/view

Reply:
I assume that happens because, towards the end of that gesture, `deltaX` becomes finally bigger than `deltaY`. That gesture starts as a vertical scroll, and ends as a horizontal one.
I'm fine keeping it like that for now, but if you don't like it we should detect somehow if the gesture starts as a horizontal pan or not

Reply:
yes, `overflow-x: hidden` will work, though I wish it could be done on the `dismissable-item` instead for better encapsulation

Reply:
Actually if the list starts scrolling, that should cancel the dismissal handling. 

If I start a drag horizontally, then I cannot scroll until I lift my finger/mouse.. even if I pan my finger all the way down vertically and then to the right. That works like Android - just try panning away one of the notifications on Android - press, move a bit left or right, then while finger down, move all the way down and then you can still dismiss by moving left and right.

I don't want to break that behavior. So basically just knowing when the list is being scrolled and then dismiss would be fine!